D1371R0.pandoc

```diff
@@ -1082,6 +1082,57 @@ it is to `vector<T>` as far as the type system is concerned. This limitation
 will likely be even bigger in a pattern matching context with the amount of
 customization points available for user-defined behavior.
 
+## Unrestricted Side Effects
+
+We considered the possibility of restricting side-effects within patterns.
+Specifically, we discussed whether modifying the value currently being matched
+in the middle of evaluation should have defined behavior.
+
+The consideration was due to potential optimization opportunities.
+
+Consider the following:
+
+```cpp
+int x = 1;
```

Comment:
I think you need something like:

```c++
int x = 1;
foo(x); //  where 'void foo(const int &)' and foo is defined in another translation unit.
```

Otherwise, the compiler can guarantee that test will not modify x because x is not referenced  between its definition and the inspect statement.

Reply:
Ah yeah. I was intending the `int x` to be a global here such that even `test()` could modify `x` even though we don't explicitly pass a reference to it. I've updated this to:

```cpp
void f(int &);  // defined in another translation unit.

int x = 1;
inspect (x) {
    0: std::cout << 0;
    1 if (f(x)): std::cout << 1;
    2: std::cout << 2;
}
```